onnxruntime/core/platform/env.h

```diff
@@ -58,13 +58,9 @@ struct ThreadOptions {
   // the main thread, which is usually set in the main executable(not controlled by onnxruntime.dll).
   unsigned int stack_size = 0;
 
-  // Thread affinity means a thread can only run on the logical processors that the thread is allowed to run on.
-  // If the vector is not empty, set the affinity of each thread to just one CPU.
-  // Index is thread index, value is CPU ID, starting from zero. For example, the first thread in the pool will be bound
-  // to the logical processor with id of affinity[0]. If the vector is empty, the thread can run on all the processors
-  // its process can run on. NOTE: When hyperthreading is enabled, for example, on a 4 cores 8 physical threads CPU,
-  // processor group [0,1,2,3] may only contain half of the physical cores.
-  std::vector<size_t> affinity;
+  // A vector hosting affinity settings for sub-threads in the threadpool
+  // Note - affinity.size() does not necessarily amount to thread_pool_size - 1
```

Comment:
>Note - affinity.size() does not necessarily amount to thread_pool_size - 1 [](http://example.com/codeflow?start=5&length=74)

Perhaps, we can clarify how thread indices map to this array. Is it still true that zero thread index would still map to [0] affinity?

Reply:
That is correct - the zero thread index will map to affinity[0].
More details:
POSIX and windows have different ways to fetch, save, and interpretation on the affinity vector.
This is because, for windows, logical processors are organized in groups and set per thread affinity with a group_id+ logic_processor_bitmastk pair, whereas POSIX simply does this by processor id(s).
That being said, for windows, we are saving group_id and logic_processor_bitmask into the affinity vector per single thread, meaning each thread will be indexed to two elements in the vector; for POSIX, it is just one.
e.g. suppose for windows, we have 3 threads in the threadpool, indexed as 0,1,2.
Then the affinity vector will have 6 elements, indexed as 0,1,2,3,4,5.
Finally, we have threads[0] mapped to affinity[0] (it is a group_id), affinity[1] (it is a logical_processor_bitmask)
thread[1] mapped to affinity[2], affnity[3];
thread[2] mapped to affinity[4], affinity[5];

Reply:
This is a good explanation, that should be put into comments.

Reply:
Added.